# Code Notebook 1: CA house prices

This notebook explores a sample dataset in Colab:

``The authors collected information on the variables using all the block groups in California from the 1990 Census. In thier sample a block group on average includes 1425.5 individuals living in a geographically compact area.'' 

[Here](https://developers.google.com/machine-learning/crash-course/california-housing-data-description) is a description of the variables.

Reference:

Pace, R. Kelley and Ronald Barry, Sparse Spatial Autoregressions,
Statistics and Probability Letters, 33 (1997) 291-297.

For each question, either run the existing code by pressing the ``play'' button by each code chunk, or add the appropriate code then run it.  The output will appear under the code. You must ``play'' the code chunks in order as some might depend on previous computations.

Use the output to answer the questions in the answer spaces.  To edit an answer space, double click it.

Text answers MUST be unique to each student but groups may share code or the numerical output of the code.


## Question 1 

Read the description in the link above.  What is a ``block group'' in this dataset?

## Answer 1: short answer.

It is a geographic area that was surveyed consisting of 1,425.5 individuals.

## Question 2 
Load the data into R and check the dimensions of the data. How many observations are there? How many variables? 

In [ ]:
housing <- read.csv("/content/sample_data/california_housing_train.csv")

In [ ]:
str(housing)

'data.frame':	17000 obs. of  9 variables:
 $ longitude         : num  -114 -114 -115 -115 -115 ...
 $ latitude          : num  34.2 34.4 33.7 33.6 33.6 ...
 $ housing_median_age: num  15 19 17 14 20 29 25 41 34 46 ...
 $ total_rooms       : num  5612 7650 720 1501 1454 ...
 $ total_bedrooms    : num  1283 1901 174 337 326 ...
 $ population        : num  1015 1129 333 515 624 ...
 $ households        : num  472 463 117 226 262 ...
 $ median_income     : num  1.49 1.82 1.65 3.19 1.93 ...
 $ median_house_value: num  66900 80100 85700 73400 65500 74000 82400 48500 58400 48100 ...


In [ ]:
dim(housing)

[1] 17000     9

## Answer 2: fill in four blanks.

There are 17000 rows and 9 columns in the housing dataset. This means there are 17000 observations of block groups from the 1990 Census and 9 pieces of information about each block group.

## Question 3 (1 point)

Obtain a summary of the variable named ``population'' in the housing dataset.  What is the largest number of people in any block group?

In [ ]:
summary(housing$population)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
      3     790    1167    1430    1721   35682 

## Answer 3: fill in one blank.

The largest block size is 35,682

## Question 4 (1 point)

Obtain a summary of the variable named ``median_income''.  What do you think the units of this variable are?  Check the description in the link to get the right answer.  Write an one sentence description of the minimum value of this variable.

In [ ]:
summary(housing$median_income)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.4999  2.5664  3.5446  3.8836  4.7670 15.0001 

## Answer 4: short answer

The minimum value of "median_income" would be 0.4999 tens of thousands of US dollars.

## Question 5 (2 points)
Calculate the ratio of bedrooms to households for each block: call this variable beds.per.house. Make a summary of this variable and use it to complete the sentence below.  

In [ ]:
beds.per.house <- housing$total_bedrooms/housing$households

summary(beds.per.house)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.3333  1.0062  1.0488  1.0973  1.1000 34.0667 

## Answer 5: Complete the sentence.

The minimum ratio of total bedrooms in a block group to total households is 0.3333.  This number is less than 1, which means there would be some households that have 0 bedrooms.

## Question 6 (2 points)

Calculate the total number of rooms that are not bedrooms for each block: store this as a variable called total_nonbedrooms.  Make a summary of this variable.

In [ ]:
total_nonbedrooms <- (housing$total_rooms - housing$total_bedrooms) 
summary(total_nonbedrooms)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
      0    1135    1682    2104    2509   32466 

## Question 7 (1 point) 

Create a new variable of your choice from the variables in the dataset.  Compute the summary of this variable.

In [ ]:
pop.household <- (housing$population/housing$household)
summary(pop.household)

    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
  0.6923   2.4326   2.8169   2.9763   3.2794 502.4615 